lib/assignment/roundrobin.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+var _ = require('lodash');
+var groupPartitionsByTopic = require('../utils').groupPartitionsByTopic;
+var debug = require('debug')('kafka-node:Roundrobin');
+var VERSION = 0;
+
+function assignRoundRobin (topicPartition, groupMembers, callback) {
+  var _members = _(groupMembers).pluck('id');
+  var members = _members.value().sort();
+  debug('members', members);
+  var assignment = _members.reduce(function (obj, id) {
+    obj[id] = [];
+    return obj;
+  }, {});
+
+  var subscriberMap = groupMembers.reduce(function (subscribers, member) {
+    subscribers[member.id] = member.subscription;
+    return subscribers;
+  }, {});
+
+  debug('subscribers', subscriberMap);
+
+  // layout topic/partitions pairs into a list
+  var topicPartitionList = _(topicPartition).map(function (partitions, topic) {
+    return partitions.map(function (partition) {
+      return {
+        topic: topic,
+        partition: partition
+      };
+    });
+  }).flatten().value();
+  debug('round robin on topic partition pairs: ', topicPartitionList);
+
+  var assigner = cycle(members);
+
+  topicPartitionList.forEach(function (tp) {
+    var topic = tp.topic;
+    while (!_.contains(subscriberMap[assigner.peek()], topic)) {
+      assigner.next();
+    }
+    assignment[assigner.next()].push(tp);
+  });
+
+  var ret = _.map(assignment, function (value, key) {
+    var ret = {};
+    ret.memberId = key;
+    ret.topicPartitions = groupPartitionsByTopic(value);
+    ret.version = VERSION;
+    return ret;
+  });
+
+  callback(null, ret);
```

Comment:
It doesn't look like any of the code here is async, but you are using callbacks, and elsewhere in the code you treat this code as async. In order to be consistent, should this be wrapped in a setImmediate? (same for the other provided assignment schema.)